x/auth/ante.go

```diff
@@ -160,14 +159,14 @@ func processSig(
 
 	// Check account number.
 	accnum := acc.GetAccountNumber()
-	if accnum != sig.AccountNumber {
+	if !simulate && (accnum != sig.AccountNumber) {
```

Comment:
Im thinking in the following functions, things might be easer to comprehend if there was just a single `simulate` or `!simulate` condition and group everything in that code block.

Reply:
ACKd